pkg/log/log_context.go

```diff
@@ -0,0 +1,31 @@
+package log
+
+import (
+	"github.com/gobuffalo/buffalo"
+	"github.com/sirupsen/logrus"
+)
+
+const logEntryKey string = "log-entry-context-key"
+
+// SetEntryInContext stores an Entry in the buffalo context
+func SetEntryInContext(ctx buffalo.Context, e Entry) {
+	ctx.Set(logEntryKey, e)
+}
+
+// EntryFromContext returns an Entry that has been stored in the buffalo context.
+// If there is no value for the key or the type assertion fails, it returns a new
+// entry from the provided logger
+func EntryFromContext(ctx buffalo.Context, l *Logger) Entry {
+	d := ctx.Data()
+	v := d[logEntryKey]
+	if v == nil {
```

Comment:
if there's no logger in the context, let's give them back a logger that does nothing. We should create a function in the log pacakge called `NoOpLogger` and call it on this line and line 27 below.

Reply:
Great suggestion. Are you thinking `NoOpLogger` would return a `Logger` instance, or should it create an actual `entry`?

Reply:
it should return a `*Logger` instance which would satisfy the Entry interface so it works here.